pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/OtherIdentifiersRule.scala

```diff
@@ -15,28 +15,24 @@ import uk.ac.wellcome.platform.merger.rules.WorkPredicates.WorkPredicate
  * - Miro IDs
  * - Digital sierra IDs
  */
+
 object OtherIdentifiersRule extends FieldMergeRule with MergerLogging {
   type FieldData = List[SourceIdentifier]
 
   override def merge(
     target: UnidentifiedWork,
     sources: Seq[TransformedBaseWork]): FieldMergeResult[FieldData] = {
-    val miroIds =
-      miroIdsRule(target, sources) getOrElse Nil
-    val physicalDigitalIds =
-      physicalDigitalIdsRule(target, sources) getOrElse Nil
-    val calmIds =
-      calmIdsRule(target, sources) getOrElse Nil
+
+    val rules = List(miroIdsRule, physicalDigitalIdsRule, calmIdsRule)
+    val mergedSources = getSourcesToMerge(rules, target, sources)
+    val ids = rules.flatMap(_(target, sources)).flatten
```

Comment:
These 3 lines are quite confusing but I think that might be helped by the proposed changes to `getSourcesToMerge`, and also by replacing the `_(target, sources)` syntax with an explicit (named) function call

Reply:
* You have the rules to apply
* The sources that those rules will apply to
* The FieldData from those rules concatenated

Would a comment work? Not sure if the confusion comes from the fact that I am using `rules` to get the fielddata and sources?

Reply:
The confusion is probably mostly due to `getSourcesToMerge` (and from the anonymous function syntax). Again, I think lists of functions are a bit hard to reason about and wonder if being explicit/repetitive would help...